### PR TITLE
feat(concepts): add transformation article

### PR DESCRIPTION
**Pull request by btroncone**
@adamlubek @Barryrowe Would you be able to review the `Get started transforming streams with map, pluck, and mapTo` article when you have a chance? Thanks!

### CODE REVIEW

File: concepts/get-started-transforming.md [hunk 1]
```diff
@@ -0,0 +1,399 @@
+# Get started transforming streams with map, pluck, and mapTo
+
+When working with observables one of the most common use cases you will
+encounter is the need to transform a stream of some value type into a stream of
+another value type. For instance, you may have an observable of click events
+that you wish to transform into an observable of objects containing just the
+`clientX` and `clientY` coordinates. Or maybe you need to extract a value from a
+stream of input events to perform a calculation or initiate a request. Or
+perhaps you just want to extract a single property from an object, like a key
+code, to perform another action down the (pipe)line. The scenarios for
```
Comment (adamlubek) [inline concepts/get-started-transforming.md]: '(pipe)line' - clever! :)

File: concepts/get-started-transforming.md [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+transforming streams are endless.
+
+In this article, we are going to learn about the most common operator used to
+transform streams, the `map` operator. We will start by taking a look at
+`Array.map` to build a general understanding of the `map` operation. Next, we
+will explore how we can apply this approach to observables by using the RxJS
+`map` operator. Finally, we will check out a few helper operators that can be
+used in place of `map` should the right scenario present itself, exploring
+common use cases along the way. Let's get started!
+
```
Comment (adamlubek) [inline concepts/get-started-transforming.md]: Maybe worth mentioning here that there are more operators that work similarly to JS array functions like filter or reduce?

File: concepts/get-started-transforming.md [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+updates from an input box, or reshaping an object, `map` will be one of the most
+used operators in your day-to-day RxJS toolbox. For scenarios where you just
+need to map to a single property, or always want to map to a constant value, you
+can also check out the [`pluck`](../operators/transformation/pluck.md) and
+[`mapTo`](../operators/transformation/mapto.md) helper operators.
+
+For a full list of transformation operators with examples, including operators
+which manage mapping to more complex values such as other observables, check out
+the [transformation operator section](../operators/transformation/README.md). We
+will explore these topics in detail in future posts!
```
Comment (adamlubek) [inline concepts/get-started-transforming.md]: Maybe worth to mention https://www.learnrxjs.io/learn-rxjs/concepts/time-based-operators-comparison here? These are transform operators too :)

**Comment by adamlubek** [comment]
Awesome article @btroncone ! really good introduction and easy to follow :)

File: concepts/get-started-transforming.md (hunk 2 above)
Comment (btroncone) [reply concepts/get-started-transforming.md]: Good idea, I'm going to add this as a tip callout in the next section. 👍

File: concepts/get-started-transforming.md (hunk 3 above)
Comment (btroncone) [reply concepts/get-started-transforming.md]: Good suggestion, need to update my wording a bit to fit this in.

**Comment by btroncone** [comment]
Thanks for the review @adamlubek!

**Comment by adamlubek** [comment]
my pleasure @btroncone ! :)

**Pull request merged**